src/Sylius/Component/Core/Promotion/Checker/TaxonRuleChecker.php

```diff
@@ -36,13 +38,13 @@ public function isEligible(PromotionSubjectInterface $subject, array $configurat
         /* @var $item OrderItemInterface */
         foreach ($subject->getItems() as $item) {
             foreach ($item->getProduct()->getTaxons() as $taxon) {
```

Comment:
Sadly, with such an approach we couldn't use early return on ``isEligible`` method ;/ I don't like nested loops as well, but in this case I think it is clear enough to leave this.